.github/workflows/electrs.yml

```diff
@@ -0,0 +1,101 @@
+name: Electrs
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  electrs-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+    - uses: actions/checkout@v2
+      if: github.event_name == 'pull_request'
+    - uses: dorny/paths-filter@v2
+      if: github.event_name == 'pull_request'
+      id: filter
+      with:
+        filters: |
+            path-filter:
+              - './!((doc|infrastructure|)/**)'
+
+  electrs-build-test-publish:
+    needs: electrs-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.electrs-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Load environment variables
+        uses: keep-network/load-env-variables@v1
+        with:
+          environment: 'ropsten'
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      # TODO: Verify if caching works
```

Comment:
It is working partially:
- it caches and restores the libraries that need to be installed in the docker container.
- it doesn't cache the actual electrs build, namely the command: `RUN cargo install --locked --path .`

The whole job takes more than 16 minutes and it's mostly because the above step is not cached.

And then I see the following error in the logs when caching takes place:
```
2021-08-31T16:44:51.7756861Z #12 writing layer sha256:bf4ba57426fbecac245e742133d13b96982991445c23dfbebd66a5c83c2e2ae4
2021-08-31T16:45:02.2228231Z time="2021-08-31T16:45:02Z" level=error msg="(*service).Write failed" error="rpc error: code = Canceled desc = context canceled" 
```